src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs

```diff
@@ -348,35 +349,16 @@ public int ReadUInt8()
         }
 
         public int ReadInt8()
-        {
-            return (sbyte)ReadUInt8();
-        }
+            => (sbyte)ReadUInt8();
 
         public int ReadUInt16()
-        {
-            int offset;
-            byte[] buffer = GetBuffer(2, out offset);
-            int i = buffer[offset + 0] + (buffer[offset + 1] << 8);
-            Advance(2);
-            return i;
-        }
+            => BitConverter.IsLittleEndian ? ReadRawBytes<ushort>() : BinaryPrimitives.ReverseEndianness(ReadRawBytes<ushort>());
 
         public int ReadInt16()
-        {
-            return (short)ReadUInt16();
-        }
+            => (short)ReadUInt16();
 
         public int ReadInt32()
-        {
-            int offset;
-            byte[] buffer = GetBuffer(4, out offset);
-            byte b1 = buffer[offset + 0];
-            byte b2 = buffer[offset + 1];
-            byte b3 = buffer[offset + 2];
-            byte b4 = buffer[offset + 3];
-            Advance(4);
-            return (((((b4 << 8) + b3) << 8) + b2) << 8) + b1;
-        }
+            => BitConverter.IsLittleEndian ? ReadRawBytes<int>() : BinaryPrimitives.ReverseEndianness(ReadRawBytes<int>());
```

Comment:
just curious, was this broken for big endian before?

Reply:
Depends on what you mean by broken. I think it should have "worked" the same on big and little endian only for integers before. That is why I have the IsLittleEndian check and ReverseEndianess for them.

In order to not change the behaviour where the binary format was "broken" (not compatible across big/little endian) there is no such code for floats etc.

Reply:
I see, so your intent is to not change observable behavior in this PR.

Reply:
Yes that is correct